gadget/layout.go

```diff
@@ -196,7 +210,15 @@ func LayoutVolumePartially(volume *Volume, constraints LayoutConstraints) (*Part
 
 // LayoutVolume attempts to completely lay out the volume, that is the
 // structures and their content, using provided constraints
-func LayoutVolume(gadgetRootDir string, volume *Volume, constraints LayoutConstraints) (*LaidOutVolume, error) {
+func LayoutVolume(gadgetRootDir, kernelRootDir string, volume *Volume, constraints LayoutConstraints) (*LaidOutVolume, error) {
+
+	// Note that the kernelRootDir may reference the running
+	// kernel if there is a gadget update or the new kernel if
+	// there is a kernel update.
+	kernelInfo, err := kernel.ReadInfo(kernelRootDir)
```

Comment:
we should probably set the info to an empty one on our own here if kernelRootDir is "" or have a flag in constraints. Because what really happens is that we try to read "/meta/kernel.yaml" which shouldn't exist, but feels fragile.